tests/js/core/widget/core/Spin/Spin.js

```diff
@@ -0,0 +1,175 @@
+				digits: 3 // 0 - doesn't complete by zeros
+			},
+			spinWidget = tau.engine.instanceWidget(spin, "Spin", options);
+
+		equal(spinWidget.option("min"), options.min, "Option min of Spin is " + options.min);
+		equal(spinWidget.option("max"), options.max, "Option max of Spin is " + options.max);
+		equal(spinWidget.option("moduloValue"), options.moduloValue,
+			"Option moduloValue of Spin is " + options.moduloValue);
+		equal(spinWidget.option("shortPath"), options.shortPath,
+			"Option shortPath of Spin is " + options.shortPath);
+		equal(spinWidget.option("duration"), options.duration,
+			"Option duration of Spin is " + options.duration);
+		equal(spinWidget.option("rollHeight"), options.rollHeight,
+			"Option rollHeight of Spin is " + options.rollHeight);
+		equal(spinWidget.option("itemHeight"), options.itemHeight,
+			"Option itemHeight of Spin is " + options.itemHeight);
+		equal(spinWidget.option("momentumLevel"), options.momentumLevel,
+			"Option momentumLevel of Spin is " + options.momentumLevel);
+		equal(spinWidget.option("scaleFactor"), options.scaleFactor,
+			"Option scaleFactor of Spin is " + options.scaleFactor);
+		equal(spinWidget.option("moveFactor"), options.moveFactor,
+			"Option moveFactor of Spin is " + options.moveFactor);
+		equal(spinWidget.option("loop"), options.loop,
+			"Option loop of Spin is " + options.loop);
+		equal(spinWidget.option("labels"), options.labels,
+			"Option labels of Spin is " + options.labels);
+		equal(spinWidget.option("digits"), options.digits,
+			"Option digits of Spin is " + options.digits);
+
+		spinWidget.destroy();
+
+	});
+
+	test("Undefined values of options Spin test", 4, function () {
+		var spin = document.getElementById("spin"),
+			options = {
+				min: undefined,
+				max: undefined,
+				duration: undefined
+			},
+			spinWidget = tau.engine.instanceWidget(spin, "Spin", options);
+
+		spinWidget.value(undefined);
+
+		equal(spinWidget.option("min"), 0, "Spin option min with undefined value check");
+		equal(spinWidget.option("max"), 0, "Spin option max with undefined value check");
+		equal(spinWidget.option("duration"), 0, "Spin option duration with undefined value check");
+		spinWidget.value(0);
+		equal(spinWidget.value(), 0, "Default option value of Spin is " + 0);
+
+		spinWidget.destroy();
+
+	});
+
+
+	test("Values Spin test", 4, function () {
+		var spin = document.getElementById("spin"),
+			options = {
+				min: 3,
+				max: 10,
+				loop: "disabled"
+			},
+			testValues = [0, 10, 100, 11],
+			spinWidget = tau.engine.instanceWidget(spin, "Spin", options);
+
+		spinWidget.value(testValues[0]);
+		equal(spinWidget.value(), testValues[0], "Spin value check (0)");
+		spinWidget.value(testValues[1]);
+		equal(spinWidget.value(), testValues[1], "Spin value check (10)");
+		spinWidget.value(testValues[2]);
+		equal(spinWidget.value(), spinWidget.option("max"), "Spin value check (100) limited to max");
+		spinWidget.option("loop", "enabled");
```

Comment:
Is it a problem to enable loop by default?

Reply:
I don't understand your comment.
This is code for testing the widget in loop situation.